### PR TITLE
Update dependencies.

### PR DESCRIPTION
**Pull request by conniey**
Updates guava and jackson dependencies.

Related to: https://github.com/Azure/azure-sdk-for-java/pull/8247

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -56,7 +56,7 @@
       <dependency>
         <groupId>com.google.guava</groupId>
         <artifactId>guava</artifactId>
-        <version>20.0</version>
+        <version>24.1.1-jre</version>
```
Comment (srnagar) [inline pom.xml]: Doesn't this also need a version update for this pom itself?

Comment (srnagar) [reply pom.xml]: Nevermind, this is a SNAPSHOT. So, don't need to bump up the version.

**Comment by jianghaolu** [comment]
@yaohaizh Would you please review the impact for management plane?

**Comment by yaohaizh** [comment]
@jianghaolu when will we publish 1.7.2 for this update?

File: pom.xml (hunk 1 above)
Comment (jianghaolu) [reply pom.xml]: The versions are automatically updated with maven release plugin.

**Pull request merged**